packages/core-mobile/app/navigation/RootScreenStack.tsx

```diff
@@ -144,6 +146,12 @@ const WalletScreenStackWithContext: FC = () => {
 
 const RootScreenStack: FC = () => {
   const walletState = useSelector(selectWalletState)
+  const { top } = useVariableSafeAreaInsets()
+  const dispatch = useDispatch()
+
+  useEffect(() => {
+    dispatch(setMainTopInset(top))
```

Comment:
how often does `top` change? also, we don't really store UI state in redux. cannot we just use the `top` value as is?

Reply:
The top value is variable depending on its parent (and device of course. This main top inset value is unavailable where it's meant to be used. You can see it being used for the `NetworkDropdown`. 

There might not have been a use before to store anything UI related in the state, but it's not a bad thing. It may or may not be necessary for the redesign.

I had to do it this way because the value comes from a hook. Otherwise I would have created a basic variable for it.

Let me see if I can replicate what the hook does without the need of a React component context and just export the variable instead.